doc/anti-patterns/long-parameter-list.md

```diff
@@ -0,0 +1,11 @@
+# Long Parameter List
+
+Long parameter list are difficult to understand, they difficult to use. The use of objects allows, change transferred data by modify an object itself. When you working with objects, you should pass just enough so that the method can get the data it needs.
+
+* Use **Replace Parameter with Method** when you can obtain data by calling a method of an object. This object can be field or other parameter.
+* **Preserve Whole Object** allows you to take a group of arguments and replace them with the object.
+* Use **Introduce Parameter Object**, if you have several arguments without logical object.
+
+***
+
+P.S. We use terminology from a book **Refactoring Improving the Design of Existing Code** by *Martin Fowler*
```

Comment:
Long parameter lists are difficult to understand and use. Wrapping them into an object allows to group parameters and change transferred data only by the object modification. When you working with objects, you should pass just enough so that the method can get the data it needs.

* Use **Replace Parameter with Method** when you can obtain data by calling a method of an object.
* **Preserve Whole Object** allows you to take a group of arguments and replace them with the object.
* Use **Introduce Parameter Object**, if you have several arguments without logically grouping object.

***

P.S. We use terminology from a book **Refactoring Improving the Design of Existing Code** by *Martin Fowler*